src/libostree/ostree-repo-commit.c

```diff
@@ -2377,6 +2409,9 @@ ostree_repo_commit_transaction (OstreeRepo                  *self,
   if (!rename_pending_loose_objects (self, cancellable, error))
```

Comment:
Can't we skip this function entirely in the case where we're writing directly into the object dir?

Reply:
Probably, but per above we'd need to make more of the transaction/staging dir conditional.